flutter/lib/src/navigation/sentry_navigation_observer.dart

```diff
@@ -0,0 +1,81 @@
+import 'package:flutter/widgets.dart';
+import 'package:sentry/sentry.dart';
+
+import 'navigation_breadcrumb.dart';
+
+/// This is a navigation observer to record navigational breadcrumbs.
+/// For now it only records navigation events and no gestures.
+///
+/// Routes can always be null and their settings can also always be null.
+/// For example, if the application starts, there is no previous route.
+/// The RouteSettings are null if a developer has not specified any
+/// RouteSettings.
+///
+/// SentryNavigationObserver must be added to the navigation observer of
+/// your used app. This is an example for [MaterialApp](https://api.flutter.dev/flutter/material/MaterialApp/navigatorObservers.html),
+/// but the integration for [CupertinoApp](https://api.flutter.dev/flutter/cupertino/CupertinoApp/navigatorObservers.html)
+/// and [WidgetsApp](https://api.flutter.dev/flutter/widgets/WidgetsApp/navigatorObservers.html) is the same.
+/// ´´´dart
+/// MaterialApp(
+///   navigatorObservers: [
+///     SentryNavigatorObserver(),
+///   ],
+///   // other parameter ...
+/// )
+/// ´´´
+///
+/// See also:
+///   - https://api.flutter.dev/flutter/widgets/RouteObserver-class.html
+class SentryNavigatorObserver extends RouteObserver<PageRoute<dynamic>> {
+  factory SentryNavigatorObserver({Hub hub = HubAdapter()}) {
+    return SentryNavigatorObserver._(hub);
+  }
+
+  SentryNavigatorObserver._(this.hub) : assert(hub != null);
+
+  final Hub hub;
+
+  @override
+  void didPush(Route<dynamic> route, Route<dynamic> previousRoute) {
+    super.didPush(route, previousRoute);
+    _addBreadcrumb(
+      type: 'didPush',
+      from: previousRoute?.settings,
+      to: route?.settings,
```

Comment:
we are not using the null safety feature yet, wondering if this would cause problems, I'd rather just do null checks till the null-safety gets stable and we do a full migration